llvm/lib/Target/RISCV/RISCVRegisterInfo.td

```diff
@@ -64,9 +64,14 @@ def sub_vrm1_6 : ComposedSubRegIndex<sub_vrm2_3, sub_vrm1_0>;
 def sub_vrm1_7 : ComposedSubRegIndex<sub_vrm2_3, sub_vrm1_1>;
 
 // GPR sizes change with HwMode.
-// FIXME: Support HwMode in SubRegIndex?
-def sub_gpr_even : SubRegIndex<-1>;
-def sub_gpr_odd  : SubRegIndex<-1, -1>;
+def sub_gpr_even : SubRegIndex<32> {
```

Comment:
Leave these as -1 so we never accidentally use 32 for non-RV32?

Reply:
Isn't 32 consistent with the size we pass to the non-HwMode part of this?

```
class GPRRegisterClass<dag regList>                                              
    : RegisterClass<"RISCV", [XLenVT, XLenFVT, i32], 32, regList> {              
  let RegInfos = XLenRI;                                                         
}    
```

Reply:
I guess if we do that then it's fine